hashes/src/sha256.rs

```diff
@@ -1034,7 +1034,7 @@ mod tests {
     }
 
     // Returns the midstate of an empty hash engine tagged with "TapLeaf".
-    fn tap_leaf_midstate() -> Midstate {
+    const fn tap_leaf_midstate() -> Midstate {
```

Comment:
This doesn't do anything interesting. A better usage would have `const TAP_LEAF_MIDSTATE: Midstate = ...;` which then gets passed around.

Reply:
Used as suggested.